### PR TITLE
fix password clean

### LINKED ISSUE #237: I am not able to clean the password

### PR DESCRIPTION
**Pull request by ramirobg94**
Fixes #237

### CODE REVIEW

File: src/screens/ConnectionSettings/screen.js [hunk 1]
```diff
@@ -177,13 +179,23 @@ const ConnectionSettings = () => {
...
+                  }
                   autoCapitalize="none"
                 />
                 <PasswordInput
                   title={t('ConnectionSettings:access_info_fields.password')}
                   onChangeText={onChangeText('password')}
-                  value={formData?.password || password}
+                  value={
+                    Objects.isEmpty(formData?.password) &&
+                    formData?.password !== ''
```
Comment (SteRiccio) [inline src/screens/ConnectionSettings/screen.js]: this condition should be always false... am I wrong?!

File: src/screens/ConnectionSettings/screen.js [hunk 2]
```diff
@@ -177,13 +179,23 @@ const ConnectionSettings = () => {
                 <Input
                   title={t('ConnectionSettings:access_info_fields.username')}
                   onChangeText={onChangeText('username')}
-                  value={formData?.username || username}
+                  value={
+                    Objects.isEmpty(formData?.username) &&
+                    formData?.password !== ''
```
Comment (SteRiccio) [inline src/screens/ConnectionSettings/screen.js]: since you have used Object.isEmpty above, why don't you use it even at line 184?

Comment (ramirobg94) [inline src/screens/ConnectionSettings/screen.js]:
```suggestion
                    formData?.username !== ''
```

File: src/screens/ConnectionSettings/screen.js (hunk 1 above)
Comment (ramirobg94) [reply src/screens/ConnectionSettings/screen.js]: a | Objects.isEmpty | !== '' |  && 
-- | -- | -- | -- |
undefined or null | True | True | True 
'' | False | False | False 
'xxxx' | False | True | False

The first row happens when you has an empty value at the beginning

Comment (SteRiccio) [reply src/screens/ConnectionSettings/screen.js]: Objects.isEmpty('') is true

Comment (ramirobg94) [reply src/screens/ConnectionSettings/screen.js]: okey i will change

**Review by SteRiccio — APPROVED** [review]

**Pull request merged**